src/torchmetrics/functional/regression/minkowski.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+from torchmetrics.utilities.exceptions import TorchMetricsUserError
+
+
+def _minkowski_distance_update(preds: Tensor, targets: Tensor, p: float) -> Tensor:
+    """Updates and returns variables required to compute Minkowski distance.
+
+    Checks for same shape of input tensors.
+
+    Args:
+        preds: Predicted tensor
+        target: Ground truth tensor
+        p: Non-negative number acting as the exponent to the errors
+    """
+    _check_same_shape(preds, targets)
+
+    if p < 0:
+        raise TorchMetricsUserError("p value must be greater than 0")
+
+    difference = torch.abs(preds - targets)
+    mink_dist_sum = torch.sum(torch.pow(difference, p))
+
+    return mink_dist_sum
+
+
+def _minkowski_distance_compute(distance: Tensor, p: float) -> Tensor:
+    """Computes Minkowski Distance.
+
+    Args:
+        distance: Sum of the p-th powers of errors over all observations
+        p: The non-negative numeric power the errors are to be raised to
+
+    Example:
+        >>> preds = torch.tensor([0., 1, 2, 3])
+        >>> target = torch.tensor([0., 2, 3, 1])
+        >>> distance_p_sum = _minkowski_distance_update(preds, target, 5)
+        >>> _minkowski_distance_compute(distance_p_sum, 5)
+        tensor(2.0244)
+    """
+    return torch.pow(distance, 1.0 / p)
+
+
+def minkowski_distance(preds: Tensor, targets: Tensor, p: float) -> Tensor:
```

Comment:
please add docstring here similar to 
https://github.com/Lightning-AI/metrics/blob/ed2249d1e250dbfc4f2e654b455f43f3d29abfb1/src/torchmetrics/functional/regression/mse.py#L57-L73